frontend/src/types/ModuleEditorTypes.ts

```diff
@@ -145,3 +148,50 @@ export interface EditContentModalProps<
   onClose: () => void;
   onSave: <T extends BlockType>(newBlock: T["content"]) => void;
 }
+
+export type FormatEnum = "bold" | "italic" | "underline";
+
+export type BlockFormatEnum = "left" | "center" | "right" | "justify";
+
+export type TextEditor = BaseEditor & ReactEditor;
+
+export type CustomText = {
+  text: string;
+  bold?: boolean;
+  italic?: boolean;
+  underline?: boolean;
+};
+
+export type ParagraphElement = {
+  type: "paragraph";
+  align?: BlockFormatEnum;
+  children: Descendant[];
+};
+
+export type LinkElement = {
+  type: "link";
+  url: string;
+  children: Descendant[];
+};
+
+export type CustomElement = ParagraphElement | LinkElement;
+
+export type LeafPropTypes = {
+  attributes: CustomText;
```

Comment:
honestly wasn't sure what the type of `attributes` should be since I didn't find any typed examples

Reply:
If it compiles without the option you can omit it probably but if not and this compiles, we can just keep it as is.

Reply:
so turns out it does not compile with out it